// @picovoice/porcupine-node/dist/builtin_keywords.js
//
// Copyright 2020-2022 Picovoice Inc.
//
// You may not use this file except in compliance with the license. A copy of the license is located in the "LICENSE"
// file accompanying this source.
//
// Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
// an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
// specific language governing permissions and limitations under the License.
//
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getBuiltinKeywordPath = exports.BuiltinKeyword = void 0;
const path = require("path");
const platforms_1 = require("./platforms");
var BuiltinKeyword;
(function (BuiltinKeyword) {
    BuiltinKeyword["ALEXA"] = "alexa";
    BuiltinKeyword["AMERICANO"] = "americano";
    BuiltinKeyword["BLUEBERRY"] = "blueberry";
    BuiltinKeyword["BUMBLEBEE"] = "bumblebee";
    BuiltinKeyword["COMPUTER"] = "computer";
    BuiltinKeyword["GRAPEFRUIT"] = "grapefruit";
    BuiltinKeyword["GRASSHOPPER"] = "grasshopper";
    BuiltinKeyword["HEY_GOOGLE"] = "hey google";
    BuiltinKeyword["HEY_SIRI"] = "hey siri";
    BuiltinKeyword["JARVIS"] = "jarvis";
    BuiltinKeyword["OK_GOOGLE"] = "ok google";
    BuiltinKeyword["PICOVOICE"] = "picovoice";
    BuiltinKeyword["PORCUPINE"] = "porcupine";
    BuiltinKeyword["TERMINATOR"] = "terminator";
})(BuiltinKeyword = exports.BuiltinKeyword || (exports.BuiltinKeyword = {}));
function getBuiltinKeywordPath(builtinKeyword) {
    const platform = (0, platforms_1.getPlatform)();
    return path.resolve(__dirname, `../resources/keyword_files/${platform}/${builtinKeyword}_${platform}.ppn`);
}
exports.getBuiltinKeywordPath = getBuiltinKeywordPath;
