// @picovoice/porcupine-node/dist/index.js
//
// Copyright 2020-2023 Picovoice Inc.
//
// You may not use this file except in compliance with the license. A copy of the license is located in the "LICENSE"
// file accompanying this source.
//
// Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
// an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
// specific language governing permissions and limitations under the License.
//
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PorcupineErrors = exports.checkWaveFile = exports.getInt16Frames = exports.getBuiltinKeywordPath = exports.BuiltinKeyword = exports.Porcupine = void 0;
const builtin_keywords_1 = require("./builtin_keywords");
Object.defineProperty(exports, "BuiltinKeyword", { enumerable: true, get: function () { return builtin_keywords_1.BuiltinKeyword; } });
Object.defineProperty(exports, "getBuiltinKeywordPath", { enumerable: true, get: function () { return builtin_keywords_1.getBuiltinKeywordPath; } });
const porcupine_1 = require("./porcupine");
exports.Porcupine = porcupine_1.default;
const wave_util_1 = require("./wave_util");
Object.defineProperty(exports, "getInt16Frames", { enumerable: true, get: function () { return wave_util_1.getInt16Frames; } });
Object.defineProperty(exports, "checkWaveFile", { enumerable: true, get: function () { return wave_util_1.checkWaveFile; } });
const PorcupineErrors = require("./errors");
exports.PorcupineErrors = PorcupineErrors;
