// @picovoice/porcupine-node/dist/pv_status_t.js
//
// Copyright 2020-2022 Picovoice Inc.
//
// You may not use this file except in compliance with the license. A copy of the license is located in the "LICENSE"
// file accompanying this source.
//
// Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
// an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
// specific language governing permissions and limitations under the License.
//
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var PvStatus;
(function (PvStatus) {
    PvStatus[PvStatus["SUCCESS"] = 0] = "SUCCESS";
    PvStatus[PvStatus["OUT_OF_MEMORY"] = 1] = "OUT_OF_MEMORY";
    PvStatus[PvStatus["IO_ERROR"] = 2] = "IO_ERROR";
    PvStatus[PvStatus["INVALID_ARGUMENT"] = 3] = "INVALID_ARGUMENT";
    PvStatus[PvStatus["STOP_ITERATION"] = 4] = "STOP_ITERATION";
    PvStatus[PvStatus["KEY_ERROR"] = 5] = "KEY_ERROR";
    PvStatus[PvStatus["INVALID_STATE"] = 6] = "INVALID_STATE";
    PvStatus[PvStatus["RUNTIME_ERROR"] = 7] = "RUNTIME_ERROR";
    PvStatus[PvStatus["ACTIVATION_ERROR"] = 8] = "ACTIVATION_ERROR";
    PvStatus[PvStatus["ACTIVATION_LIMIT_REACHED"] = 9] = "ACTIVATION_LIMIT_REACHED";
    PvStatus[PvStatus["ACTIVATION_THROTTLED"] = 10] = "ACTIVATION_THROTTLED";
    PvStatus[PvStatus["ACTIVATION_REFUSED"] = 11] = "ACTIVATION_REFUSED";
})(PvStatus || (PvStatus = {}));
exports.default = PvStatus;
